channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -539,7 +539,9 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
      */
     private _xhrSender(payload: string[], isAsync: boolean) {
         const xhr = new XMLHttpRequest();
-        xhr[DisabledPropertyName] = true;
+        if (CoreUtils.isExtensible(xhr)) {
```

Comment:
The same problem can also occur if the object has bee frozen as well, in which case the the isExtensible() would still return true.

Reply:
The approach I took in my quick hack for this was to just wrap in a try / catch. My review is also fixing a lot of other stuff as well.

https://github.com/microsoft/ApplicationInsights-JS/pull/1218/files#diff-9e41298c570c651c39f1223741cb5cb7R553-R557

Reply:
@MSNev Cool.  And you've added tests around this as well.  Do you think I should just close this PR and wait for your PR to merge and get released?

Reply:
Your call, but if my PR addresses your issues then perhaps close it and we can add the helpers leter.

Reply:
I'm also planning on getting my PR in today and releasing a version 2.5.2 with the fixes